lib/src/dropdown/dropdown.component.ts

```diff
@@ -0,0 +1,503 @@
+            this.currentFocused = -1;
+        }
+        this.searchText = event.target.value;
+    }
+
+    /** Function containing the select dropdown item logic */
+    optionItemSelected(item: Item): void {
+        if (!this.multi) {
+            const newItem = { ...item };
+            this.handleOnChange(newItem);
+            this.open = false;
+        } else {
+            const currentList = (this.selectedValue as Array<Item>) ? (this.selectedValue as Array<Item>) : [];
+            const index = currentList.findIndex((e: Item) => e.key === item.key);
+            if (index === -1) {
+                const newItem = { ...item };
+                const newList = [...currentList, newItem];
+                this.handleOnChange(newList);
+            } else {
+                const newList = currentList.filter((e: Item) => e.key !== item.key);
+                this.handleOnChange(newList);
+            }
+        }
+        this.handleFocus();
+    }
+
+    /** The native event function that runs when the dropdown button is clicked */
+    handleClickToggle(event: any): void {
+        if (!this.disabled) {
+            this.open = !this.open;
+        }
+    }
+
+    /** Function containing the select all button logic */
+    handleSelectAll(): void {
+        if (this.allSelected) {
+            this.handleOnChange([]);
+        } else {
+            this.handleOnChange(this.list);
+        }
+        this.handleFocus();
+    }
+
+    // HELPERS ================================
+    /** Returns the appropriate title for different situations and component types */
+    getTitleLabel() {
+        if (this.uniqueList && this.uniqueList.length === 0) {
+            return "Empty";
+        }
+        if (this.selectedList && this.selectedList.length > 0) {
+            if (this.allSelected) {
+                return `All selected (${this.selectedList.length})`;
+            }
+            if (this.multi) {
+                if (this.selectedList.length === 1) {
+                    return this.selectedList[0].label;
+                }
+                return this.selectedList
+                    .map((item, index) => {
+                        if (index > 0) {
+                            return ` ${item.label}`;
+                        } else {
+                            return item.label;
+                        }
+                    })
+                    .toString();
+                // return this.selectedList.length + " Selected"; // TODO should be like this example: 1st Item, 2nd Item... (+2)
+            }
+            return (this.selectedValue as Item).label;
+        }
+
+        return this.placeHolder && this.placeHolder.length ? this.placeHolder : "Select ...";
+    }
+
+    handleItemOnMouseMove(event: any, index: number): void {
+        this.currentFocused = index;
+        this.shouldFocus = false;
+    }
+
+    handleItemOnClick(event: any, index: number, item: DisplayItem): void {
```

Comment:
if you can please add the types for all the instances of `event: any` in this file